### DISCUSSION TITLE
Do I need to use a custom server in order to implement db access and authentication?

### DISCUSSION THREAD

**Discussion by pedrorolo** [alephjs/aleph.js#435] [Q&A] [answered]
Hi! 

When I mean custom server I mean this: https://alephjs.org/docs/advanced-features/custom-server

I am being a bit put off by the idea of having to resort to such hack in order to implement such universal needs. Do you know if this is indeed the case? Do you have any suggestion regarding what libs should I have a look at in order to sort out these two functionalities?

(P.S. I am new to Deno and to the TS world, and programmed much more in Ruby than in JS - with which I am unfamiliar with the ecosystem.)

**Comment by shadowtime2000** [comment, marked as answer]
Disclaimer: I'm going to be real I've been out of touch with the project so you can probably take what I say with a grain of salt

For traditional access to databases you will have to use a custom server given that Aleph is really built on being serverless, however iirc there are a lot of serverless database tools that allow you to access the database through REST queries so you probably want to research something like that, or you could set up your own server to mediate such requests to a traditional database of your own hosting.
